File: solutions/php/resistor-color-trio/1/ResistorColorTrio.php
<?php

declare(strict_types=1);

class ResistorColorTrio
{
    public function label(array $colors): string
    {
        $tr = [
            'black'  => 0,
            'brown'  => 1,
            'red'    => 2,
            'orange' => 3,
            'yellow' => 4,
            'green'  => 5,
            'blue'   => 6,
            'violet' => 7,
            'grey'   => 8,
            'white'  => 9,
        ];
        $conv = [
            ['giga', 1e9],
            ['mega', 1e6],
            ['kilo', 1e3],
        ];

        $value = (
            $tr[$colors[0]] * 10 + $tr[$colors[1]]
        ) * 10 ** $tr[$colors[2]];
        $unit = 'ohms';

        foreach ( $conv as $c ) {
            if ( $value > $c[1] ) {
                $unit = $c[0] . $unit;
                $value /= $c[1];
                break;
            }
        }
        $string = "$value $unit";

        return $string;
    }
}

File: solutions/lua/resistor-color-trio/1/resistor-color-trio.lua
return {
    decode = function(c1, c2, c3)
        local rcc = {
            ['black' ] = 0,
            ['brown' ] = 1,
            ['red'   ] = 2,
            ['orange'] = 3,
            ['yellow'] = 4,
            ['green' ] = 5,
            ['blue'  ] = 6,
            ['violet'] = 7,
            ['grey'  ] = 8,
            ['white' ] = 9,
        }
        local value = (rcc[c1] * 10 + rcc[c2]) * math.pow(10, rcc[c3])
        local unit = 'ohms'
        local metric_prefix = {
            {9, 'giga'},
            {6, 'mega'},
            {3, 'kilo'},
        }
        for i = 1, #metric_prefix do
            base = math.pow(10, metric_prefix[i][1])
            name = metric_prefix[i][2]
            if value > base then
                value = value / base
                unit = name .. unit
                break
            end
        end
        
        return value, unit
    end
}
